File: dsa/array_string_obj/longestSubstring.js
let s = "abcabcbb";
let str = s.split("");
let tempStr = "";

for (let i = 0; i < str.length; i++) {
  let substr = "";
  for (let j = i; j < str.length; j++) {
    if (substr.includes(str[j])) {
      break;
    } else {
      substr += str[j];
    }
  }
  if (tempStr.length < substr.length) {
    tempStr = substr;
  }
}
console.log(tempStr);

File: dsa/array_string_obj/longestSubstring.py
s = "abcabcbb"
temp_str = ""

for i in range(len(s)):
    substr = ""
    for j in range(i, len(s)):
        if s[j] in substr:
            break
        substr += s[j]
    if len(substr) > len(temp_str):
        temp_str = substr

print(temp_str)
